crates/iota-genesis-builder/src/stardust/types/address_swap_split_map.rs

```diff
@@ -0,0 +1,172 @@
+// Copyright (c) 2024 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use std::collections::HashMap;
+
+use iota_sdk::types::block::address::Address;
+use iota_types::base_types::IotaAddress;
+
+type OriginAddress = Address;
+type Destination = (IotaAddress, u64, u64);
+
+#[derive(Clone, Debug, Default)]
+pub struct AddressSwapSplitDestinations {
+    destinations: Vec<Destination>,
+}
+
+impl AddressSwapSplitDestinations {
+    /// Returns a mutable iterator that filters destinations with both
+    /// tokens_target and tokens_timelocked_target values > 0
```

Comment:
Nit: This is rather an iterator over mutable values, not a mutable iterator. Also I see only `tokens_target` being used in the filtered.
```suggestion
    /// Iterate over mutable destinations filtered by `tokens_target > 0`.
```